### ISSUE TITLE
form-builder.js dependencies not initializing Alpaca

### ISSUE THREAD

**Issue by tomsjer** [gitana/alpaca#266]
Hello, I'm working on an interesting project that involves generating surveys/forms with Alpaca.js, storing them on a CouchDB and getting them to work offline (on tablets running Chrome for Android) and stay in sync with PouchDB.

My issue is that I can´t get  Alpaca to initialize (and thus triggering the postRender callback that fires the modal)  whenever I [comment the line that delete the dependencies](https://github.com/tomsjer/alpaca/blob/master/site/demos/form-builder/form-builder.js#L413) retrieved by [**Field#getOptionsForSchema**](https://github.com/tomsjer/alpaca/blob/master/src/js/Field.js#L2011).

Have you got any suggestions/workarounds or fixes? I'll appreciate it.
And thanks for this plugin, it's GREAT!

**Comment by tomsjer** [comment]
Hi, finally I identified the problem.

In the **Field class  #getOptionsForSchema** the function that filled the *dataSoure* expected 2 arguments: *field,callback*. The real argument (only one) that is passed to this function is the callback function, so, **field** was assigned to a function and **callback** to undefined,  hence never firing up and failing to initialize Alpaca.

Also, there is a property **schemaParent** in the condition wich I couldnt find anywhere in Alpaca's source, ¿maybe from an old version?

Summing up, after rewriting the *dataSource function* and not deleting the dependencies property of the field in the form-builder now I am able to list and set the dependencies for each field in the designer! :)

I hope it helps someone, [heres the code](https://github.com/tomsjer/alpaca/commit/b08fd5f42d277761c6411a0a05367afdd187dcd1)

**Comment by tomsjer** [comment]
Hi @Rmathusuthanan, sorry about the broken links.
This is the original code that wasn't working for me:
```javascript
"dataSource": function (field, callback) {
    if (field.parent && field.parent.schemaParent && field.parent.schemaParent.parent) {
        for (var key in field.parent.schemaParent.parent.childrenByPropertyId) {
            if (key != field.parent.schemaParent.propertyId) { // jshint ignore:line
                field.selectOptions.push({
                    "value": key,
                    "text": key
                });
            }
        }
    }
    if (callback) {
        callback();
    }
}
```
Heres what I modified:
```javascript
"dataSource": function (callback) { // Only recieving callback as argument!

    if (self.parent && !self.parent.parent) { // Reference to the field
        for (var key in self.parent.childrenByPropertyId) {
            if(key !== self.propertyId){
                
                var txt = (self.parent.childrenByPropertyId[key].options.label) ? self.parent.childrenByPropertyId[key].options.label : key;
            
                this.selectOptions.push({
                    "value": key,
                    "text": txt
                });
            }
        }
    }

    if (callback) {
        callback();
    }
},
```
I guess that the function [is this one](https://github.com/gitana/alpaca/blob/master/src/js/Field.js#L2445) (?) IDK, it's been a while.
Cheers!

**Comment by Rmathusuthanan** [comment]
@tomsjer  Thanks for your reply 

I tried that code but I am getting "None" Value in dependencies.

**Comment by Rmathusuthanan** [comment]
Hi Tom,

 You have any a idea about the alpacajs wizard in form builder. its
possible to create wizard steps  in dynamically in  form builder ?

Thanks
Mathusuthanan
